client/components/payment-activity/hooks.ts

```diff
@@ -0,0 +1,132 @@
+/**
+ * External dependencies
+ */
+import { useState } from 'react';
+import { __ } from '@wordpress/i18n';
+import moment from 'moment';
+
+interface DateRange {
+	/** The name of the date range preset. e.g. last_7_days */
+	preset_name: string;
+	/** The date range start datetime used to calculate transaction data, e.g. 2024-04-29T16:19:29 */
+	date_start: string;
+	/** The date range end datetime used to calculate transaction data, e.g. 2024-04-29T16:19:29 */
+	date_end: string;
+}
```

Comment:
There is a lot of overlap between the `DateRange` and the presets. It may be easier to follow we reduce these to a single interface?

Example:
```ts
interface DateRangePreset {
	presetName: string;
	dateStart: moment.Moment;
	dateEnd: moment.Moment;
  /** e.g. Last 7 days */
  displayValue: string;
  /** e.g. June 1 - June 7 */
  hint: string;
}
```

This would also allow us to simplify the `useState` and `onChange` handlers – they'd only need to worry about selecting a preset name (`last_7_days`).

Reply:
> This would also allow us to simplify the useState and onChange handlers – they'd only need to worry about selecting a preset name (last_7_days).

This sounds great … though I have no idea whether combining the types/structs is a good idea or a stretch.

Reply:
Currently, we have `DateRange` which requires dates as a string, and the equivalent `DateRangePreset` which stores dates as a `moment` object.

It works well as-is, so happy to leave this as a future iteration.